File: ios/Packages/SolennixCore/Sources/SolennixCore/Models/Payment.swift
import Foundation

public struct Payment: Codable, Identifiable, Sendable, Hashable {
    public let id: String
    public let eventId: String
    public let userId: String
    public let amount: Double
    public let paymentDate: String
    public let paymentMethod: String
    public var notes: String?
    public let createdAt: String
    
    // MARK: - Init
    
    public init(
        id: String,
        eventId: String,
        userId: String,
        amount: Double,
        paymentDate: String,
        paymentMethod: String,
        notes: String? = nil,
        createdAt: String
    ) {
        self.id = id
        self.eventId = eventId
        self.userId = userId
        self.amount = amount
        self.paymentDate = paymentDate
        self.paymentMethod = paymentMethod
        self.notes = notes
        self.createdAt = createdAt
    }
}

File: android/core/model/src/main/java/com/creapolis/solennix/core/model/Payment.kt
package com.creapolis.solennix.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Payment(
    @SerialName("id") val id: String = "",
    @SerialName("event_id") val eventId: String = "",
    @SerialName("user_id") val userId: String = "",
    val amount: Double,
    @SerialName("payment_date") val paymentDate: String,
    @SerialName("payment_method") val paymentMethod: String,
    val notes: String? = null,
    @SerialName("created_at") val createdAt: String = ""
)

@Serializable
data class CreatePaymentRequest(
    @SerialName("event_id") val eventId: String,
    @SerialName("user_id") val userId: String,
    val amount: Double,
    @SerialName("payment_date") val paymentDate: String,
    @SerialName("payment_method") val paymentMethod: String,
    val notes: String? = null
)
